src/s1reader/s1_burst_slc.py

```diff
@@ -622,3 +623,76 @@ def swath_name(self):
         '''Swath name in iw1, iw2, iw3.'''
         return self.burst_id.split('_')[1]
 
+    @property
+    def thermal_noise_lut(self):
```

Comment:
I think `thermal_noise_lut` would be better as class members instead as a `@property`. Identical computations would be performed each time this property is called without any change to the parameters, so I think this would be more efficient as a class member that's computed one-time. I also think the computations here would also be better suited as a member function of `BurstNoise` that takes shape as a parameter given `burst_noise` members are extensively utilized.

The changes would be something like:

s1_burst_slc.py
```
...
@dataclass(frozen=True)
class Sentinel1BurstSlc:
    ...
    thermal_noise_lut: np.array
```

s1_reader.py
```
...
def burst_from_xml()
    ...
    thermal_noise_lut = burst_noise.compute_thermal_noise_lut(shape)
    ...
    bursts[i] = Sentinel1BurstSlc(..., thermal_noise_lut, ...)
```

s1_annotation.py
```
...
@dataclass
class BurstNoise:
    ...
    def computer_thermal_noise_lut(self, shape):
        <computations highly similar to below>
...
```

Let me know what you think.

Reply:
@LiangJYu Very good point. I agree with your idea that it makes more sense to calculate the LUT only once.

My concern is that we turn on or off the thermal noise correction by user input in `runconfig`. I expect we will apply the correction most of the time, but we will need to turn it off when requested (for testing, comparison, etc.).
In that case, calculating the LUT is not necessary.

We can just set `thermal_noise_lut` to `None` when not applying the correction, and use it when initializing `Sentinel1BurstSlc` in `burst_from_xml()`. However, `burst_from_xml()` needs information whether user has opted in/out the correction. Maybe `s1-reader` need to access `runconfig`, but not sure if that is a good idea in terms of the independency of `s1-reader` to the workflow implementations.

I know you are not a fan of load-on-access pattern like below, but I think it could be one of the ways to optimize. Looking forward to hear about your opinion.

For your information: in our current RTC workflow, we use `thermal_noise_lut` only once when processing a burst.

s1_burst_slc.py
```python
class Sentinel1BurstSlc:
...
    _thermal_noise_lut : np.array = None

    @property
    def thermal_noise_lut(self):
        if _thermal_noise_lut is None:
            _thermal_noise_lut = <Compute the LUT>

        return _thermal_noise_lut

```


Regarding the location of the LUT computation:
> I also think the computations here would also be better suited as a member function of `BurstNoise` that takes shape as a parameter given `burst_noise` members are extensively utilized.

In user perspective, I prefer that we do not need to care about the LUT's shape by making it same as the corresponding burst's. But I agree that the LUT computation makes more sense as the method of `BurstNoise`
So I think I can move the computation part into `BurstNoise`, keep the method 
`thermal_noise_lut` inside `Sentinel1BurstSlc`, and let that method call `BurstNoise.computer_thermal_noise_lut()` by providing it with the burst's shape.

Reply:
Given the very significant memory usage mentioned [here](https://github.com/seongsujeong/s1-reader/pull/6#pullrequestreview-1093763303), I withdraw my suggestion of making `thermal_noise_lut` a class attribute. 

Having the `Sentinel1BurstSlc.thermal_noise_lut` property call `BurstNoise.compute_thermal_noise_lut(burst.shape)` sounds good to me.

Reply:
@LiangJYu Thanks for the suggestion. I've made modification to the structure.